src/Data/Array/ST.purs

```diff
@@ -146,12 +141,10 @@ thaw = copyImpl
 foreign import toAssocArray """
   function toAssocArray(arr) {
     return function(){
-      var as = [];
-      var i = -1;
-      var n = arr.length;
-      while(++i < n) {
+      var n = arr.length >>> 0;
+      var as = new Array(n);
+      for (var i = 0 >>> 0; i < n; i++)
```

Comment:
Just to ensure unsigned values. Probably useless in this case (not in the case of peekSTArray, where it saves a comparison). Do you want me to remove this one?